src/platforms/windows/windowscommons.cpp

```diff
@@ -67,22 +70,43 @@ QString WindowsCommons::tunnelConfigFile() {
   return QString();
 }
 
+// static
 QString WindowsCommons::tunnelLogFile() {
-  QStringList paths =
-      QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
+  static QString tunnelLogFilePath = getTunnelLogFilePath();
+  return tunnelLogFilePath;
+}
 
-  for (const QString& path : paths) {
-    QDir dir(path);
-    if (!dir.exists()) {
-      continue;
-    }
+// static
+QString WindowsCommons::getProgramFilesPath() {
+  wchar_t* path = nullptr;
 
-    QDir vpnDir(dir.filePath(VPN_NAME));
-    if (!vpnDir.exists()) {
-      continue;
-    }
+  if (SUCCEEDED(
+          SHGetKnownFolderPath(FOLDERID_ProgramFiles, 0, nullptr, &path))) {
+    auto guard = qScopeGuard([&] { CoTaskMemFree(path); });
+    return QString::fromWCharArray(path);
+  }
+  return QString();
```

Comment:
From my understanding, if `SHGetKnownFolderPath` fails to locate the directory, it'll return an empty string. Would it make more sense to eliminate line 88 `return QString();`?

Reply:
SHGetKnownFolderPath can fail  -- it can return an E_FAIL or other error code, causing the SUCCEEDED() macro to return false. We need to return an empty QString() on line 88 to handle that case.